zuc/docker-mojito
# Refactor init system

Refactor init system:

 * Remove requirements for extra bash script for env substitution
 * Use the `openjdk` docker image (instand of `java` that's an alias)
 * Remove `-XX:MaxPermSize=128m` that's ignored
 * Use the container limit for CPU and RAM using `-XX:+UnlockExperimentalVMOptions` and `-XX:+UseCGroupMemoryLimitForHeap`

To test the correct behaviour:

1. Build docker images

       docker build -t mojito:full -f full/Dockerfile full/
       docker build -t mojito:cli-only -f cli-only/Dockerfile cli-only/

1. Start a database (use a different tab)

       docker run -it --rm \
           --name sampledb \
           -e MYSQL_RANDOM_ROOT_PASSWORD=true \
           -e MYSQL_DATABASE=mojito \
           -e MYSQL_USER=mojito \
           -e MYSQL_PASSWORD=mypass \
            mariadb:10.1

1. Check that the database is ready

       docker run -it --rm \
           --name debugdb \
           --link sampledb \
           mariadb:10.1 \
           mysql \
           -hsampledb \
           -pmypass \
           -umojito

1. Start the web instance

       docker run -it --rm \
           --name mojito-web \
           --link sampledb \
           -p 8080:8080 \
           -e MOJITO_DB_NAME=mojito \
           -e MOJITO_DB_USER=mojito \
           -e MOJITO_DB_PASS=mypass \
           -e MOJITO_DB_HOST=sampledb \
           -e MOJITO_DB_PORT=3306 \
           mojito:full

1. Check that the CLI work as expected

       docker run -it --rm \
           --name mojito-cli \
           --link mojito-web \
           -e MOJITO_API_HOST=mojito-web \
           -e MOJITO_API_PORT=8080 \
           -e MOJITO_API_SCHEME=http \
           -e MOJITO_API_USER=admin \
           -e MOJITO_API_PASS=ChangeMe \
           mojito:cli-only \
           user-create  -gn sample -un sample -sn sample -cn sample

1. Try to log-in with user `sample` and related password insert before
